x-pack/plugins/security_solution/server/lib/detection_engine/fleet_integrations/api/get_installed_integrations/installed_integration_set.ts

```diff
@@ -33,10 +33,59 @@ interface PackageInfo extends InstalledPackageBasicInfo {
 export const createInstalledIntegrationSet = (): IInstalledIntegrationSet => {
   const packageMap: PackageMap = new Map<string, PackageInfo>([]);
 
+  const addPackage = (fleetPackage: PackageListItem): void => {
+    if (fleetPackage.type !== 'integration') {
+      return;
+    }
+    if (fleetPackage.status !== 'installed') {
+      return;
+    }
+
+    const packageKey = `${fleetPackage.name}`;
+    const existingPackageInfo = packageMap.get(packageKey);
+
+    if (existingPackageInfo != null) {
+      return;
+    }
+
+    // Actual `installed_version` is buried in SO, root `version` is latest package version available
+    // SO should be available as PackageListItem should be InstalledRegistry<RegistrySearchResult>
+    const installedPackageVersion = fleetPackage.savedObject?.attributes?.install_version;
+    const latestPackageVersion = fleetPackage.version;
+
+    // Policy templates correspond to package's integrations.
+    const packagePolicyTemplates = fleetPackage.policy_templates ?? [];
+
+    const packageInfo: PackageInfo = {
+      package_name: fleetPackage.name,
+      package_title: fleetPackage.title,
+      package_version: installedPackageVersion ?? latestPackageVersion,
+
+      integrations: new Map<string, InstalledIntegrationBasicInfo>(
+        packagePolicyTemplates.map((pt) => {
+          const integrationTitle: string =
+            packagePolicyTemplates.length === 1 && pt.name === fleetPackage.name
+              ? fleetPackage.title
+              : pt.title;
```

Comment:
If it's a package with a single integration of the same name, we get the title from the package, not the integration. These values seem to be different for a lot of packages, `Microsoft 365` is one of them.